solution/index.js

```diff
@@ -0,0 +1,310 @@
+                break;
+            case 'submit-add-in-progress':
+                savedData['in-progress'].unshift(liValue);
+                break;
+            case 'submit-add-done':
+                savedData['done'].unshift(liValue);
+                break;
+            case 'to-do-tasks':
+                savedData['todo'].unshift(liValue);
+                break;
+            case 'in-progress-tasks':
+                savedData['in-progress'].unshift(liValue);
+                break;
+            case 'done-tasks':
+                savedData['done'].unshift(liValue);
+                break;
+        }
+    
+
+     saveToLocal(savedData);
+}
+    
+function editTask(e) //edits task value on dbl click
+{   
+    const target= e.target;
+    const mouseEvent= e;
+    if(target.classList.contains("task"))
+    {
+        const li= target;
+        const oldValue = li.textContent;
+        const input = document.createElement("input");
+        input.placeholder = oldValue;
+        li.replaceWith(input);
+        input.classList.add("task");
+        input.focus(); //right when a user dblClicks the input focuses
+        input.onblur= (e)=>{
+            if(input.value!=="")
+            {
+                li.textContent = input.value;
+                input.replaceWith(li); //replaces the li with the new value
+                const section = li.closest("section");
+                const buttonId = section.querySelector("button").id;
+                SaveData(buttonId, li.textContent);
+                const key= getKeyValue(mouseEvent);
+                deleteData(key, oldValue)
+            }
+            else{alert("Must enter Value")}
+         }
+    }
+
+}
+
+function deleteData(key, value) //delets value from local data
+{
+    const dataArray= getLocal;
+    const valueIndex = dataArray[key].indexOf(value);
+    dataArray[key].splice(valueIndex,1);
+    saveToLocal(dataArray);
+}
+function getKeyValue(e)
+{
+    const liItem= e.target;
+    const containingSec = liItem.closest("section");
+    const h2Text = containingSec.querySelector("h2").textContent;
+    if(h2Text.includes("To Do"))
+    {
+        return "todo"
+    }
+    if(h2Text.includes("In Progress"))
+    {
+        return "in-progress"
+    }
+    if(h2Text.includes("Done"))
+    {
+        return "done"
+    }
+}
+function duplicateLi({target}) //take existing li items content and creates new li item
+{
+    const liContent= target.textContent;
+    const newTask = document.createElement("li");
+    newTask.textContent = liContent;
+    newTask.classList.add("task");
+    return newTask;
+}
+function changeSect(e) //changes section based on alt + num entered
+{
+    if(e.target.classList.contains("task"))
+    {
+        e.target.style.backgroundColor = "lightgray";
+        let keysPressed= {};
+        const mouseEvent= e;
+        document.onkeydown= (event)=>{
+            keysPressed[event.key]= true;
+            if(keysPressed["Alt"] && event.key === "1")
```

Comment:
There is an easy way to look at a keyboard event and find out if the alt key was pressed during that event.
This is not the way :)